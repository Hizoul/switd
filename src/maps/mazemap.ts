import Map, { experimentChoices } from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMapNoSplits = () => {
  const testMapNoSplits = new Map(15, 15)
  testMapNoSplits.spawnThreshold = 14
  testMapNoSplits.experimentType = experimentChoices.shortestPathOnly
  const start = new Street(0, 0, testMapNoSplits)
  testMapNoSplits.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.right)
  const s3 = s2.extend(direction.right)
  const s4 = s3.extend(direction.right)
  const s5 = s4.extend(direction.right)
  const s6 = s5.extend(direction.right)
  const s7 = s6.extend(direction.right)
  const s8 = s7.extend(direction.right)
  const s9 = s8.extend(direction.right)
  const s10 = s9.extend(direction.right)
  const s11 = s10.extend(direction.right)
  const s12 = s11.multiExtend(direction.down, 4)
  const s13 = s12.multiExtend(direction.left, 1)

  const s2a = s2.extend(direction.down)
  const s2b = s2a.extend(direction.down)
  const s2ca = s2b.multiExtend(direction.right, 3)
  const s2cb = s2ca.multiExtend(direction.right, 4)
  const s2d = s2cb.multiExtend(direction.down, 2)
  const s2e = s2d.multiExtend(direction.down, 5)

  s13.addIntersection(s2d)

  const w3 = s2ca.extend(direction.down)
  const w4 = w3.extend(direction.down)
  const w5 = w4.multiExtend(direction.right, 3)

  w5.addIntersection(s2d)

  const t1 = new Tower(4, 6, testMapNoSplits)
  t1.damagableAnts = 4
  t1.damageAmount = 60
  t1.towerRange = 2
  testMapNoSplits.addTile(t1)

  return testMapNoSplits

}

export default createtestMapNoSplits()
export { createtestMapNoSplits }
