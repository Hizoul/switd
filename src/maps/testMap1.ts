import Map from "../logic/map"
import Street, { direction } from "../logic/street"

const createtestMap1 = () => {
  const testMap1 = new Map()
  const start = new Street(0, 0, testMap1)
  testMap1.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)

  const s31 = s2.extend(direction.down)
  const s311 = s31.extend(direction.down)
  const s312 = s311.extend(direction.down)
  const s313 = s312.extend(direction.right)
  const s314intersection = s313.extend(direction.right)

  const s32 = s2.extend(direction.right)
  const s321 = s32.extend(direction.right)
  const s322 = s321.extend(direction.down)
  const s323 = s322.extend(direction.down)
  s323.addIntersection(s314intersection)
  const s4 = s314intersection.extend(direction.down)
  const s5 = s4.extend(direction.down)
  const s6 = s5.extend(direction.right)
  return testMap1
}

export default createtestMap1()
export { createtestMap1 }
