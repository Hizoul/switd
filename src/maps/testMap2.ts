import Map from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMap2 = () => {
  const testMap2 = new Map()
  testMap2.spawnThreshold = 14
  const start = new Street(0, 0, testMap2)
  testMap2.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)

  const s31 = s2.extend(direction.down)
  const s311 = s31.extend(direction.down)
  const s312 = s311.extend(direction.down)
  const s313 = s312.extend(direction.right)
  const s314 = s313.extend(direction.right)
  const s315intersection = s314.extend(direction.right)

  const s32 = s2.extend(direction.right)
  const s321 = s32.extend(direction.right)
  const s322 = s321.extend(direction.right)
  const s323 = s322.extend(direction.down)
  const s324 = s323.extend(direction.down)
  s324.addIntersection(s315intersection)
  const s4 = s315intersection.extend(direction.down)
  const s5 = s4.extend(direction.down)
  const s6 = s5.extend(direction.right)

  const t1 = new Tower(2, 2, testMap2)
  testMap2.addTile(t1)
  const t2 = new Tower(3, 3, testMap2)
  testMap2.addTile(t2)
  const t3 = new Tower(5, 5, testMap2)
  testMap2.addTile(t3)

  return testMap2
}

export default createtestMap2()
export { createtestMap2 }
