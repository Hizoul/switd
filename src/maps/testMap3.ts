import Map from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMap3 = () => {
  const testMap2 = new Map()
  testMap2.spawnThreshold = 14
  const start = new Street(0, 0, testMap2)
  testMap2.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)

  const s31 = s2.extend(direction.down)
  const s311 = s31.extend(direction.down)
  const s312 = s311.extend(direction.down)
  const s313 = s312.extend(direction.right)
  const s314 = s313.extend(direction.right)
  const s315intersection = s314.extend(direction.right)

  const s32 = s2.extend(direction.right)

  testMap2.intersectionList.push({
    beginning: s31, target: s32
  })

  const s321 = s32.extend(direction.right)
  const s322 = s321.extend(direction.right)
  const s323 = s322.extend(direction.down)
  const s324 = s323.extend(direction.down)
  s324.addIntersection(s315intersection)
  const s4 = s315intersection.extend(direction.down)
  const s5 = s4.extend(direction.down)
  const s6 = s5.extend(direction.right)

  const t2 = new Tower(2, 3, testMap2)
  testMap2.addTile(t2)
  const t3 = new Tower(5, 5, testMap2)
  testMap2.addTile(t3)

  return testMap2
}

export default createtestMap3()
export { createtestMap3 }
