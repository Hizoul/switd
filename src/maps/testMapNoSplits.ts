import Map from "../logic/map"
import Street, { direction } from "../logic/street"
import Tower from "../logic/tower"

const createtestMapNoSplits = () => {
  const testMapNoSplits = new Map()
  testMapNoSplits.spawnThreshold = 14
  const start = new Street(0, 0, testMapNoSplits)
  testMapNoSplits.addTile(start)
  const s1 = start.extend(direction.right)
  const s2 = s1.extend(direction.down)
  const s3 = s2.extend(direction.down)
  const s4 = s3.extend(direction.right)
  const s5 = s4.extend(direction.right)
  const s6 = s5.extend(direction.right)
  const s7 = s6.extend(direction.down)
  const s8 = s7.extend(direction.down)
  const s9 = s8.extend(direction.left)
  const s10 = s9.extend(direction.left)
  const s11 = s10.extend(direction.left)
  const t1 = new Tower(2, 1, testMapNoSplits)
  testMapNoSplits.addTile(t1)
  const t2 = new Tower(3, 3, testMapNoSplits)
  testMapNoSplits.addTile(t2)
  const t3 = new Tower(4, 5, testMapNoSplits)
  testMapNoSplits.addTile(t3)
  return testMapNoSplits
}

export default createtestMapNoSplits()
export { createtestMapNoSplits }
