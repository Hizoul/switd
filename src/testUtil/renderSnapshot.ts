import * as render from "preact-render-to-json"

const renderSnapshot = (comp: any, msg: string) => {
  const tree = render(comp)
  expect(tree).toMatchSnapshot(msg)
}

export default renderSnapshot
