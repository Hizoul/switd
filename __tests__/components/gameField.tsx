import * as React from "react"
import GameFieldDisplay from "../../src/components/gameField"
import { createtestMap1} from "../../src/maps/testMap1"
import { createtestMapNoSplits } from "../../src/maps/testMapNoSplits"
import noOp from "../../src/noOp"
import renderSnapshot from "../../src/testUtil/renderSnapshot"
it("Expect GameField to be able to render the field", () => {

  const testedMap = createtestMap1()
  renderSnapshot(<GameFieldDisplay fieldToRender={testedMap} />, "no ants yet")
  testedMap.spawnNewAnt(5)
  renderSnapshot(<GameFieldDisplay fieldToRender={testedMap} />, "five unmoved ants")
  testedMap.registerComponentToUpdate({setState: noOp})
  testedMap.processTick()
  testedMap.spawnNewAnt(5)
  renderSnapshot(<GameFieldDisplay fieldToRender={testedMap} />, "five ants at stage 2 5 more newly added")
  testedMap.registerComponentToUpdate({setState: noOp})
  testedMap.processTick()
  renderSnapshot(<GameFieldDisplay fieldToRender={testedMap} />, "five ants at stage 3 5 more at 1")

  const testMap2 = createtestMapNoSplits()
  renderSnapshot(<GameFieldDisplay fieldToRender={testMap2} />, "no ants yet")
  testMap2.spawnNewAnt(12)
  renderSnapshot(<GameFieldDisplay fieldToRender={testMap2} />, "12 ants at start")
  testMap2.processTick()
  testMap2.processTick()
  testMap2.processTick()
  testMap2.spawnNewAnt(3)
  testMap2.processTick()
  renderSnapshot(<GameFieldDisplay fieldToRender={testMap2} />, "12 ants at 4  3 at 1")
})
