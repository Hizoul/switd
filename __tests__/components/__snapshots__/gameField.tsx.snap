// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`12 ants at 4  3 at 1 1`] = `
<div
  class="gameField"
>
  <div
    class="street isStart"
    style={
      Object {
        "height": "76.8px",
        "left": "0px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      3
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      7
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street isTarget"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="tower"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div />
  </div>
  <div
    class="tower"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div />
  </div>
  <div
    class="tower 3"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "384px",
        "width": "102.4px",
      }
    }
  >
    <div>
      3
    </div>
  </div>
</div>
`;

exports[`12 ants at start 1`] = `
<div
  class="gameField"
>
  <div
    class="street isStart"
    style={
      Object {
        "height": "76.8px",
        "left": "0px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      12
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street isTarget"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="tower"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div />
  </div>
  <div
    class="tower"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div />
  </div>
  <div
    class="tower 3"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "384px",
        "width": "102.4px",
      }
    }
  >
    <div>
      3
    </div>
  </div>
</div>
`;

exports[`five ants at stage 2 5 more newly added 1`] = `
<div
  class="gameField"
>
  <div
    class="street isStart"
    style={
      Object {
        "height": "76.8px",
        "left": "0px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      5
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      5
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "384px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street isTarget"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
</div>
`;

exports[`five ants at stage 3 5 more at 1 1`] = `
<div
  class="gameField"
>
  <div
    class="street isStart"
    style={
      Object {
        "height": "76.8px",
        "left": "0px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      5
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      5
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "384px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street isTarget"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
</div>
`;

exports[`five unmoved ants 1`] = `
<div
  class="gameField"
>
  <div
    class="street isStart"
    style={
      Object {
        "height": "76.8px",
        "left": "0px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      5
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "384px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street isTarget"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
</div>
`;

exports[`no ants yet 1`] = `
<div
  class="gameField"
>
  <div
    class="street isStart"
    style={
      Object {
        "height": "76.8px",
        "left": "0px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "384px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street isTarget"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "460.79999999999995px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
</div>
`;

exports[`no ants yet 2`] = `
<div
  class="gameField"
>
  <div
    class="street isStart"
    style={
      Object {
        "height": "76.8px",
        "left": "0px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "0px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "153.6px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="street isTarget"
    style={
      Object {
        "height": "76.8px",
        "left": "102.4px",
        "top": "307.2px",
        "width": "102.4px",
      }
    }
  >
    <div>
      0
    </div>
  </div>
  <div
    class="tower"
    style={
      Object {
        "height": "76.8px",
        "left": "204.8px",
        "top": "76.8px",
        "width": "102.4px",
      }
    }
  >
    <div />
  </div>
  <div
    class="tower"
    style={
      Object {
        "height": "76.8px",
        "left": "307.20000000000005px",
        "top": "230.39999999999998px",
        "width": "102.4px",
      }
    }
  >
    <div />
  </div>
  <div
    class="tower 3"
    style={
      Object {
        "height": "76.8px",
        "left": "409.6px",
        "top": "384px",
        "width": "102.4px",
      }
    }
  >
    <div>
      3
    </div>
  </div>
</div>
`;
